<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>

  <remap from="scan" to="base_scan"/>


  <node pkg="exploration_ros" name="slam" type="slam_node" output="screen" args="-timeUpdate 0 -angularUpdate 0.5 -linearUpdate 0.5 -scanTopic scan -type 0">

    <param name="resolution" value="0.025" /> <!-- doc="resolution of the matching grid"  -->
    <param name="maxScore" value="0.15" /> <!-- doc="score of the matcher, the higher the less matches" /> -->
    <param name="kernelRadius" value="0.2" /> <!-- doc="radius of the convolution kernel" /> -->
    <param name="minInliers" value="5" /><!--  doc="min inliers"/> -->
    <param name="windowLoopClosure" value="10" /> <!-- doc="sliding window for loop closures"  /> -->
    <param name="inlierThreshold" value="2" /> <!-- doc="inlier threshold" /> -->
    <param name="localizationAngularUpdate" value="0.5" /> <!-- doc="angular rotation interval for updating the graph, in radians" default="0.7853975"/> -->
    <param name="localizationLinearUpdate"  value="0.5" /> <!-- doc="linear translation interval for updating the graph, in meters" default="0.25" /> -->
    <param name="localizationTimeUpdate"  value="0" /> <!-- doc="interval in time for updating the robot pose in the map, in seconds" default="0" /> -->
    <param name="typeExperiment" value="0" /> <!-- doc="0 if stage simulation or 1 if real robot experiment"  default="0"  /> -->
    <param name="laserFrameName" value="base_laser_link" /> <!-- doc="name of the laser TF frame"/> -->
    <param name="odometryFrame" value="odom" /> <!-- doc="odometry ROS topic"/> -->
    <param name="scanTopic" value="base_scan" /> <!-- value="scan" doc="scan ROS topic"/> -->
    <param name="occupancyTopic" value="map" /> <!-- doc="occupancy grid ROS topic"/> -->

  </node>


<!--
<include file="$(find exploration_ros)/move_base_config/slam_gmapping.xml"/>
-->

<node pkg="exploration_ros" name="explorer" type="exploration_action_node" output="screen">  
    <param name="mapFrame" value="map" /> <!--  doc="TF mapFrame for the robot" --> 
    <param name="baseFrame" value="base_link" />  <!-- doc="TF base Frame for the robot" -->
    <param name="laserFrame" value="base_laser_link" /> <!-- doc="TF laser Frame for the robot" -->
    <param name="scanTopic" value="scan" /> <!--  doc="laser scan ROS topic" -->
    <param name="pointsTopic" value="points" /> <!-- doc="frontier points ROS topic" -->
    <param name="markersTopic" value="markers" /> <!-- doc="frontier centroids ROS topic" -->
    <param name="regionSize" value="15" /> <!-- doc="minimum size of a frontier region" -->
    <param name="exploredArea" value="10" /> <!-- doc="minimum number of visible frontier points before aborting a goal" -->
    <param name="lambda" value="1.25" /> <!-- doc="distance decay factor for choosing next goal" -->
    <param name="mc" value="0.5" /> <!-- doc="Lower distance limit to consider 2 goals as the same, in meters" -->
    <param name="Mc" value="5.0" /> <!-- doc="Max distance at which a centroid is considered if there are also closer ones, in meters"  -->
    <param name="nc" value="10" /> <!-- doc="Maximum number of centroids considered during each search for a goal" -->
    <param name="iter" value="-1" /> <!-- doc="Number of plans to be computed. -1 means infinite" -->
</node> 

<node pkg="exploration_ros" name="action_client" type="exploration_action_client" output="screen"/> 

<node pkg="stage_ros" type="stageros" name="stageros" args="$(find exploration_ros)/stagefiles/hospital.world" respawn="false" >
  <param name="base_watchdog_timeout" value="0.2"/>
</node>


<node pkg="rviz" type="rviz" name="visualizer" output="screen" args="-d $(find exploration_ros)/config.rviz"/>


<include file="$(find exploration_ros)/move_base_config/move_base.xml"/>



</launch>

